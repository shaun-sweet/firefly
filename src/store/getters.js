export default {
  user (state) {
    return {
      displayName: state.displayName,
      email: state.email,
      id: state.id
    }
  },

  homesList (state) {
    return Object.keys(state.homes).map(homeId => {
      return {
        label: state.homes[homeId].nickname,
        value: homeId
      }
    })
  },

  selectedHome (state) {
    return state.selectedHome
  },

  isLoadingInitialState (state) {
    return state.appState.isLoadingInitialState
  },

  defaultHome (state) {
    return Object.keys(state.homes).filter(home => state.homes[home].role === 'owner')[0]
  },

  devicesViewList (state, getters) {
    if (!getters.isLoadingInitialState) {
      const devicesViewList = state.homes[state.selectedHome].devicesViewList
      const devicesArray = Object.keys(devicesViewList).map((deviceId) => {
        const device = devicesViewList[deviceId]
        // this is what auto complete fuzzy finder search feature needs to display the properties
        device.stamp = device.primaryStateStatus
        device.label = device.alias
        return device
      })
      return devicesArray.filter((device) => device.export_ui)
    }
    return []
  }

  // notifications (state, getters) {
  //   const selectedHome = state.homes[getters.selectedHome]
  //   return Object.keys(selectedHome.notifications).map(id => selectedHome.notifications[id])
  // },

  // events (state, getters) {
  //   const selectedHome = state.homes[getters.selectedHome]
  //   return Object.keys(selectedHome.events).map(id => selectedHome.events[id])
  // }
}
