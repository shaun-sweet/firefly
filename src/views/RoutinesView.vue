<template>
  <div class="layout-padding routines-view">
    <div class="row">
      <routine 
        v-if="routine.export_ui"
        :key="routine.ff_id"
        v-for="routine in routines"
        :name="routine.alias"
        :routineId="routine.ff_id"
        :command="routine.command"
        :icon="routine.icon"
        v-on:activateRoutine="activateRoutine"
      />
    </div>
  </div>
</template>

<script>
import { Toast } from 'quasar'
import Routine from '@/Routine'
import { mapGetters } from 'vuex'
import { executeRoutine } from 'src/firebase'

export default {
  components: {
    Routine
  },
  beforeMount() {
    this.$store.dispatch('getRoutines')
  },
  computed: {
    ...mapGetters([
      'routines',
      'selectedHome'
    ])
  },
  methods: {
    activateRoutine ({ msg, routineId, command }) {
      const homeId = this.selectedHome
      executeRoutine({ routineId, homeId, command }, (err) => {
        if (err) {
          console.error(err)
        } else {
          Toast.create.positive({
            html: msg || 'Routine Activated',
            icon: 'ion-ios-checkmark'
          })
        }
      })
    }
  }
}
</script>

<style lang="stylus" scoped>

</style>
