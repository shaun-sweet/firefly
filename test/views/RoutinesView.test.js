import { mount, createLocalVue } from 'vue-test-utils'
import Vuex from 'vuex'
import RoutinesView from 'src/views/RoutinesView.vue'
import getters from 'src/store/getters'
import {initialState, testState} from '../fixtures/state'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('RoutinesView', () => {
  let wrapper;
  let store;
  let state;
  let spy;
  let actions;
  beforeEach(() => {
    spy = jest.fn()
    state = {...testState}
    actions = {  getRoutines: spy  }
    store = new Vuex.Store({
      state,
      getters,
      actions
    })
    wrapper = mount(RoutinesView,{ store, localVue });
  })
  it('should render the RoutinesView correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should call dispatch getRoutines beforeCreate', () => {
    expect(spy).toHaveBeenCalled()
  });

})
