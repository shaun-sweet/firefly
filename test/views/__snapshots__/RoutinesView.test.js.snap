// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutinesView should render the RoutinesView correctly 1`] = `
<div
  class="layout-padding routines-view"
>
  <div
    class="row"
  >
    <!---->
    <!---->
  </div>
</div>
`;
